annotation-processor/src/main/java/org/fulib/fx/FulibFxProcessor.java

```diff
@@ -273,8 +294,68 @@ private boolean isController(TypeMirror typeMirror) {
         return declaredType.asElement().getAnnotation(Controller.class) != null;
     }
 
-    private boolean isProvider(TypeMirror typeMirror) {
-        return typeMirror.toString().startsWith("javax.inject.Provider");
+    /**
+     * Checks if an element is an event element (annotated with {@link ControllerUtil#EVENT_ANNOTATIONS}).
+     *
+     * @param element The element to check
+     * @return True if the element is an event element
+     */
+    private boolean isEventElement(Element element) {
+        for (Class<? extends Annotation> eventAnnotation : ControllerUtil.EVENT_ANNOTATIONS) {
+            if (element.getAnnotation(eventAnnotation) != null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if the given method overrides another event method.
+     *
+     * @param method       The method to check
+     * @param eventMethods A map of all event methods of the superclasses
+     */
+    private void checkOverrides(ExecutableElement method, Map<String, List<ExecutableElement>> eventMethods) {
+        TypeElement element = (TypeElement) processingEnv.getTypeUtils().asElement(method.getEnclosingElement().asType());
+
+        // If no parent class is found, the method cannot override anything
+        if (element.getSuperclass().getKind() == TypeKind.NONE) {
+            return;
+        }
+
+        // Check if one of the parent classes has the method
+        if (eventMethods.containsKey(method.getSimpleName().toString())) {
```

Comment:
Small refactoring: this should have early return, and the simple name tostring should be in a variable instead of being repeated